clients/src/main/java/org/apache/kafka/common/protocol/ApiKeys.java

```diff
@@ -217,7 +217,8 @@ public Struct parseResponse(short version, ByteBuffer buffer) {
 
     public final Schema[] requestSchemas;
     public final Schema[] responseSchemas;
-    public final boolean requiresDelayedAllocation;
+    public final boolean requestRequiresDelayedAllocation;
+    public final boolean responseRequiresDelayedAllocation;
```

Comment:
Some Response objects, like JoinGroupResponse, have fields directly backed by ByteBuffer. So even when the network buffer has been read into an object, we still need to keep it. This is what we're trying to keep track here. 
So deallocation of these can only be done once the Response object is not used anymore, instead of directly once the Response has been built (for Response without ByteBuffer fields)

Reply:
I understood the purpose of the flag, I was just nitpicking the name. Allocation happens immediately, right? I thought possibly `responseRequiresManualDeallocation` would be more accurate.